patches/tModLoader/Terraria/ModLoader/MenuLoader.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.Audio;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader.Default;
+using Terraria.UI.Chat;
+
+namespace Terraria.ModLoader
+{
+	internal static class MenuLoader
+	{
+		internal static ModMenu currentModMenu;
+
+		private static readonly List<ModMenu> moddedMenus = new List<ModMenu>();
+
+		private static List<ModMenu> lastAvailableMenus = new List<ModMenu>();
+
+		internal static List<ModMenu> AvailableMenus => moddedMenus.Where(m => m.IsAvailable).ToList();
+
+		internal static string lastUsedModMenuName = $"ModLoader/{nameof(MenutML)}";
+
+		internal static void Add(ModMenu modMenu) {
+			moddedMenus.Add(modMenu);
+		}
+
+		private static void OffsetModMenu(int offset) {
+			var menus = AvailableMenus;
```

Comment:
what happens if a menu goes unavailable? eg a holiday menu as the clock hits midnight. Crash?

Reply:
If the ModMenu is unavailable while selected, it will remain selected until the player selects a different menu (at which point it will no longer be selectable). In addition, crashes while reloading are preventing by setting the menu to null when an unload occurs.